.github/workflows/reusable-context.yml

```diff
@@ -97,6 +97,8 @@ jobs:
       run: python Tools/build/compute-changes.py
       env:
         GITHUB_DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+        CCF_TARGET_REF: ${{ github.base_ref || github.event.repository.default_branch }}
```

Comment:
What does CCF stand for here?

And why do we need two different options? Is this being executed in different contexts?

Reply:
compute changed files, just a unique prefix.

We always want the most recent commit of each branch & to compare against the HEAD of the default branch. For `pull_request` events, `github.sha` is a fabricated merge commit, so we use `.head.sha`. For `push`and `workflow_dispatch` events, `pull_request.head.sha` is unset and `github.sha` does what we want, hence the fallback.

It's similar for the target/base reference, `pull_request` events define the property but `push` and `workflow_dispatch` events don't, so we use the default branch name as a fallback.